### PR TITLE
Surface max_receive_count and approximate_receive_count

### PR DESCRIPTION
**Pull request by micahswab**
So we can do cool things like only alert when the message is about to get tossed on DLQ.

### CODE REVIEW

**Comment by ArtskydJ** [comment]
I think we'll want to cast that string to a number.

We'll also want to add a test for the new functionality.  Maybe we could add something like this to `index.test.mjs`
```js
assert.type(result_of_receive.approximate_receive_count, `number`)
```

**Comment by ArtskydJ** [comment]
LGTM.  Would probably be good to wait for @TehShrike to review too.

**Comment by TehShrike** [comment]
So, so far this library basically just exposes the SQS API, and doesn't implement any other special/opinionated behavior.

SQS doesn't expose a `GetQueueRedrivePolicy` action, so it seems out of place next to all the other methods – I'd expect an API more like this:

```js
const { 
	dead_letter_target_arn, 
	max_receive_count 
} = await sqs.get_queue_attributes(queue_url, [ `redrive_policy` ])
```

and maybe something like this for receive message

```js
const { attributes } = await sqs.receive_message(queue_url, { attributes: [ `approximate_receive_count` ] })

attributes.approximate_receive_count // => some number
```

it could look different from that, the main point being I think there's value in having this library staying nothing more than a sanity layer between us and SQS.

We don't want to have to deal with navigating XML hierarchies or manually building those wacky attribute properties or converting from snake to pascal case, but otherwise, it's nice that the methods all map pretty directly to the SQS API, and you can read the SQS docs to see what everything does.

File: index.mjs [hunk 1]
```diff
@@ -34,6 +36,21 @@ const concat_text = element => element.children.filter(({ type }) => type === `t
...
+
+	return Object.fromEntries(attributes.map((attribute) => {
+
+		const name = read_text_from_descendant(attribute, `Name`)
+		const text_value = read_text_from_descendant(attribute, `Value`)
+		const int_value = parseInt(text_value)
+
+		return [
+			make_snake_case(name),
+			Number.isNaN(int_value) ? text_value : int_value
```
Comment (TehShrike) [inline index.mjs]: `parseInt` is a bit too lenient to trust, probably – `parseInt('1 or 2 seems fine')` evaluates to `1`.

Maybe 

```suggestion
		const text_value = read_text_from_descendant(attribute, `Value`)
		const is_all_digits = /^\d+$/.test(text_value)

		return [
			make_snake_case(name),
			is_all_digits ? parseInt(text_value, 10) : text_value
```

instead?

It feels bad to have to do duck-typey coercion, but that's probably preferable to just returning strings

**Review by TehShrike — COMMENTED** [review]
other than the parseInt fiddling, this seems good to me

wide shot of a 🦞 eating the number 3, dramatic lighting

**Pull request merged**